src/Compilers/Core/Portable/RuleSet/RuleSet.cs

```diff
@@ -142,7 +144,7 @@ private RuleSet GetEffectiveRuleSet(HashSet<string> includedRulesetPaths)
                 var effectiveRuleset = ruleSet.GetEffectiveRuleSet(includedRulesetPaths);
 
                 // Apply the includeAction on this ruleset.
-                effectiveRuleset = effectiveRuleset.WithEffectiveAction(ruleSetInclude.Action);
+                effectiveRuleset = effectiveRuleset.WithEffectiveAction(ruleSetInclude.Action)!;
```

Comment:
>! [](start = 94, length = 1)

How do you know?

Reply:
`WithEffectiveAction` only returns `null` for `ReportDiagnostic.Suppress` and for `default` case in `switch`. But we checked for `Suppress` case above, and the `switch` is complete on this enum (ie. the `default` will not be reachable).
That said, I'll convert this to an assertion, since it's a pretty indirect check.